### DISCUSSION TITLE
Normalized feature extraction

### DISCUSSION THREAD

**Discussion by SinaRanjkeshzade** [blue-yonder/tsfresh#857] [General]
Hi, 
Is it possible to get extracted features of tsfresh for a Time-Series in normalized form? 
Even by normalizing the input data, output features are not normalized. 
If not, what is the best distance criteria to compute the distance between two tsfresh extracted feature vectors? 
I'd rather use cosine distance but when features are not normalized, some features with large values have more effect on distance .

**Comment by nils-braun** [comment]
Hi @SinaRanjkeshzade,
I am not an expert on the distance metric (maybe @kempa-liehr or @MaxBenChrist can comment), but I can give you the information, that there is no normalization of the resulting feature vector possible. The reason is, that our feature calculators are quite different (e.g. the range of possible values of the "mean" calculator depends on the data, whereas a auto-correlation coefficient might always be between -1 and 1). 

I do not know your use case, but what you could try is to extract the features from a reasonable large number of examples in your data set and find the range of possible values of the features for your domain using this data. With that, you could then normalize (or if needed: clip) new feature extraction results.

**Reply by kempa-liehr** [reply]
Hi @SinaRanjkeshzade,
I agree with @nils-braun. In general, the normalisation of time-series features is problem-specific and needs to be incorporated into the design of the machine learning pipeline.  

However, you might want to consider unsupervised time-series feature selection prior to your normalisation: https://doi.org/10.1109/JSEN.2021.3084970

Cheers,
Andreas

**Reply by SinaRanjkeshzade** [reply]
Thank's for your response. 
Although it's not the best idea, but instead of computing distance by cosine method or similar ones which are sensitive to the features' range, I used `abs(y-x)/x`. "x" is what I expect to happen and "y" is what happened in real. 
This formula scales the distance of most features to [0,1].
